manager/integration/tests/test_ha.py

```diff
@@ -836,3 +836,107 @@ def test_rebuild_with_inc_restoration(
+    3. Do cleanup for the backupstore.
+    4. Create a pod with a volume and wait for pod to start.
+    5. Write data to the pod volume and get the md5sum.
+    6. Create a backup for the volume.
+    7. Randomly delete some data blocks of the backup, which will lead to
+       all replica restore failures later.
+    8. Restore a volume from the backup.
+    9. Wait for the volume restore in progress by checking if:
+       9.1. `volume.restoreStatus` shows the related restore info.
+       9.2. `volume.conditions[restore].status == True &&
+            volume.conditions[restore].reason == "RestoreInProgress"`.
+       9.3. `volume.ready == false`.
+    10. Wait for the restore volume Faulted.
+    11. Check if `volume.conditions[restore].status == False &&
+        volume.conditions[restore].reason == "RestoreFailure"`.
+    12. Check if `volume.ready == false`.
+    13. Check if the auto-salvage is disabled even if the setting is enabled.
+    """
+    pass
+
+
+@pytest.mark.skip
+def test_single_replica_restore_failure(
+        client, core_api, volume_name, pod_make):  # NOQA
+    """
+    [HA] Test if one replica restore failure will lead to the restore volume
+    becoming Degraded, and if the restore volume is still usable after
+    the failure.
+
+    Notice that this case is similar to test_rebuild_with_restoration().
+    But the way to fail the replica is different.
+    test_rebuild_with_restoration() directly crash the replica process
+    hence there is no error in the restore status.
+
+    1. Enable auto-salvage.
+    2. Set the a random backupstore.
+    3. Do cleanup for the backupstore.
+    4. Create a pod with a volume and wait for pod to start.
+    5. Write data to the pod volume and get the md5sum.
+    6. Create a backup for the volume.
+    7. Restore a volume from the backup.
+    8. Wait for the volume restore start by checking if:
+       8.1. `volume.restoreStatus` shows the related restore info.
+       8.2. `volume.conditions[restore].status == True &&
+            volume.conditions[restore].reason == "RestoreInProgress"`.
+       8.3. `volume.ready == false`.
+    9. Find a way to fail just one replica restore.
+       e.g. Use iptable to block the restore.
+    10. Wait for the restore volume Degraded.
+    11. Check if `len(restore_volume.replicas) == 2`.
+        Then check if the volume condition and ready status keep unchanged.
+    12. Wait for the volume restore complete and detached.
+    13. Check if
+        13.1. there is no rebuilt replica for the restored volume
+              before reattachment.
+        13.2. `volume.ready == false`
+        13.3. `volume.conditions[restore].status == False &&
+              volume.conditions[restore].reason == ""`.
+    14. Create PV/PVC/Pod for the restored volume and wait for the pod start.
+    15. Check if the restored volume is state `degraded`
+        after the attachment.
+    16. Wait for the rebuild complete and the volume becoming healthy.
```

Comment:
We can add an additional step here

Verify `len(restore_volume.replicas) == 3`.

Reply:
Sure, but I am not sure if I need to put this kind of implementation details in the skeleton.
The reason I mention some detailed checks in the skeleton is that some fields are newly introduced after the fix and some fields are critical for the state check.

Reply:
If volume suppose to have three replicas and it's healthy, that means all three replicas should be healthy. So in this case, there is no need to mention it since we will make sure the volume is healthy.